### ISSUE TITLE
AbstractStatBind can't get the stat_key properly.

### ISSUE THREAD

**Issue by Sironix** [Zennyth/EnhancedStat#10]
After rigorous testing it seems like the stat_key variable is always null.
Testing out with printing the values of the  variables all got printed out properly except for the stat_key one.
And well without being able to communicate the values of the stat nodes with each other's nor scripts makes the whole project unusable at the moment.

**Comment by Zennyth** [comment]
I just checked with my project example and (AbstractStatBind) seems to be working as expected inside and outside of the editor : 
![image](https://github.com/Zennyth/EnhancedStat/assets/51298462/11fa10c0-53db-42f7-b1b4-e30c6bc20237)

Could you explain to me what did you do and what you want to do with the StatBinders ?

**Comment by Sironix** [comment]
Have you tried it in game?
I'm just trying to use it as the wiki explains it, even the character example that is in the example folder doesn't work.
The stat bind nodes simply don't seem to get the stat key from the stat manager nodes.
Maybe you have an updated or outdated version of the addon or its dependecies?

**Comment by Zennyth** [comment]
Yes I did and the demo was working without any problem.

**Comment by Sironix** [comment]
wow thats amazing because some of the required nodes didn't exist the last time I tried. (I'll download the update you did a few hours ago to see if they came back I guess or if the example got updated.)

Anyway. 
what seems to happen is that the button in the stat key getter of the abstract stat bind doesn't seem to work.
when creating a new abstract bind node, the export buttons of "get statmanager", "get node target" and "get property to bind" seem to work, but the "select stat of the stat manager" doesn't.
It appears empty, with no options till you select a different node and then come back.
![Empty Stat crop](https://github.com/Zennyth/EnhancedStat/assets/85960918/f4ae98dc-2635-424c-b3b5-0d187fc5ea11)

![updated stat crop](https://github.com/Zennyth/EnhancedStat/assets/85960918/36ff1ba1-dea4-4c25-b5c9-30513c088103)

but afterwards when trying the game it shows the error "abstract_stat_bind.gd:30 @ bind(): Stat has not been configured properly."
and looking at the statbind node in the remote inspector it displays some isssues.
![inspector](https://github.com/Zennyth/EnhancedStat/assets/85960918/e2a26aa7-fa77-4cf7-bc8c-438a92a5b542)
so... I dunno what to tell you.

**Comment by Zennyth** [comment]
Okay I understand you're issues now : 
- The first one is that the buttons are not working when you instantiate the statbind for the first time and to get around that just select another node and reselect the statbind afterward. You should be able to see the buttons and your stats. I'm gonna fix that rn :)
- The second is that you need to put the stat bind after the statsmanager and not before or as a child. It is because the statbind needs the statsmanager and therefore needs to be ready after the statsmanager. I'll try to find a way to able that but for now just put the statbind just after 

Tell me if this was it :)

**Comment by Sironix** [comment]
ah, I was following the example in your character example, how those statbinds were always childrens of other nodes.
I'll try it out.

**Comment by Sironix** [comment]
Still didn't work, lol..

**Comment by Zennyth** [comment]
Alright I did reproduced your problem by having only one stat, the editor says that speed is bound but in fact it isn't. Try to add another stat to your statmanager and open the dropdown and select speed.

**Comment by Sironix** [comment]
oki, now I get "Invalid operands 'int' and 'Object' in operator '=='."
abstract stat bind in line 34

**Comment by Zennyth** [comment]
But honeslty this doesn't matter cause I'll post an update now on the rework of those statbinders. To resolve you need to choose a StatValueBind cause you only need the value of this stat not the entire object.

**Comment by Sironix** [comment]
also unrelated issue but it looks like having two stats in a component with the same class really freaks out the statbind node.
(says that the statkey already existed in the map)
afterwards tried with speed and health and got the error in the previous message.

**Comment by Zennyth** [comment]
yeah because you're supposed to create a class for each stat, the id of the stat is its type.

**Comment by Sironix** [comment]
> But honeslty this doesn't matter cause I'll post an update now on the rework of those statbinders. To resolve you need to choose a StatValueBind cause you only need the value of this stat not the entire object.

oki will wait for it.

**Comment by Sironix** [comment]
> yeah because you're supposed to create a class for each stat, the id of the stat is its type.

yeah, wanted to point it out just in case you didn't knew about it.

**Comment by Zennyth** [comment]
I've pushed the update on stat binders, just replace StatValueBind by StatBind and choose the option Bind value inside the editor.

**Comment by Zennyth** [comment]
Keep me up to date if you've figured it out, but it should resolve both issues and be easier to use.

**Comment by Sironix** [comment]
new statbind node also doesn't work?
" Resource file not found: res:// (expected type: )
  res://addons/enhanced_common/scripts/editor/properties/option_properties.gd:42 - Cannot call method 'get_image' on a null value."

**Comment by Sironix** [comment]
I don't really know how it works for you but it doesn't work for me.

**Comment by Zennyth** [comment]
I'm stupid, try to add an icon to your stat class ...

**Comment by Sironix** [comment]
also when creating the node it seems to try to load a file that doesn't exist anymore  
Failed loading resource: res://addons/enhanced_stat/scripts/core/stats/bind/abstract_stat_value_bind.gd. Make sure resources have been imported by opening the project in the editor at least once.
will try adding an icon to the class

**Comment by Zennyth** [comment]
yes thats because I've deleted it, search for it in this file .godot/global_script_class_cache.cfg and delete the section about it.

**Comment by Sironix** [comment]
oki

**Comment by Zennyth** [comment]
Alright, I've pushed the fix on enhance_common for the empty icon.

**Comment by Sironix** [comment]
neato, won't try it today tho. 
got back into the project with 127 errors due to the class_names cache deletion.
so I'm gonna fix that and then once I cool down I may check this plugin again.

**Comment by Sironix** [comment]
btw the new version can't be loaded.
  modules/gdscript/gdscript.cpp:2775 - Failed to load script "res://addons/enhanced_stat/plugin.gd" with error "Parse error"

**Comment by Zennyth** [comment]
Of course I’ve tested it with my example project  but this one is tricky cause I needed to restart the editor to notice the error as it was running just fine. I also tested it in my personal project.

I also need to remind you that this project is in alpha because it’s version for is 0.0.1 and that this is my first serious open source project.

I’ll fix the problem later in the day. I’m sorry for all your troubles but this was bound to happen…
